morpheus/messages/tensor_memory.py

```diff
@@ -31,11 +31,75 @@ class TensorMemory(MessageData, cpp_class=_messages.TensorMemory):
     Parameters
     ----------
     count : int
-        Number of inference inputs.
-    inputs : typing.Dict[str, cupy.ndarray]
-        Inference inputs to model.
+        Length of each tensor contained in `tensors`.
+    tensors : typing.Dict[str, cupy.ndarray]
+        Collection of tensors uniquely identified by a name.
 
     """
     count: int
 
-    tensors: typing.Dict[str, cp.ndarray] = dataclasses.field(default_factory=dict, init=False)
+    tensors: typing.Dict[str, cp.ndarray] = dataclasses.field(default_factory=dict,
+                                                              repr=False,
+                                                              compare=False,
+                                                              hash=False)
+
+    def __init__(self, count: int, tensors: typing.Dict[str, cp.ndarray] = {}):
```

Comment:
You dont want to use `{}` as the default argument. This sets `self._tensors` by reference which means it can be shared by multiple classes. Set the default to `None` and then `self._tensors = {} if tensors is None else tensors`

Reply:
Ugh, good catch!